src/librustc/session/mod.rs

```diff
@@ -158,6 +158,9 @@ impl Session {
     pub fn fatal(&self, msg: &str) -> ! {
         panic!(self.diagnostic().fatal(msg))
     }
+    pub fn fatal_panic_only(&self) -> ! {
```

Comment:
maybe `fatal_with_no_message` or something is a better name? I am a bit confused by `fatal_panic_only`